### PR TITLE
Add “cwd” to DecentralizedComputation

### PR DESCRIPTION
**Pull request by swashcap**
Asana task: https://app.asana.com/0/37131652076106/94154588782863

* adds a `cwd` property to DecentralizedComputation
* adds some model tests
* makes ComputationRegistry add `cwd` to `DecentralizedComputation` instances when appropriate (when `instance.setup` is a string)

### CODE REVIEW

**Comment by swashcap** [comment]
Shouldn’t conflict with #15 hopefully.

File: src/models/decentralized-computation.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+'use strict';
+
```
Comment (cdaringe) [inline src/models/decentralized-computation.js]: whoops thxxxx

File: src/models/decentralized-computation.js [hunk 2]
```diff
@@ -11,18 +13,19 @@ const util = require('util');
  * respective environments.  This can be used to help publishers assert
  * that their computation is complete and accurate
  */
-function DecentralizedComputation(attrs) {
+function DecentralizedComputation() {
```
Comment (cdaringe) [inline src/models/decentralized-computation.js]: whoops thxxxx

File: src/models/decentralized-computation.js [hunk 3]
```diff
@@ -11,18 +13,19 @@ const util = require('util');
...
 DecentralizedComputation.schema = Object.assign({
+    cwd: joi.string().min(2),
     name: joi.string().min(2).required(),
     setup: joi.alternatives().try(
         joi.string(),
         joi.func()
     ),
     repository: joi.object().keys({
-        url: joi.string().min(9)
+        url: joi.string().uri().min(9)
```
Comment (cdaringe) [inline src/models/decentralized-computation.js]: got all the tricks huh!  [such great joi h4ckz](https://github.com/hapijs/joi/blob/v8.0.1/API.md#stringurioptions)

File: src/services/classes/computation-registry.js [hunk 4]
```diff
@@ -225,7 +225,35 @@ function ComputationRegistry(options) {
...
+     * …or a JavaScript function:
+     *
+     *     {
+     *       //...
+     *       setup: function(cb) {
+     *         fs.doSomethingWithFiles('param', cb);
+     *       },
+     *     }
+     *
+     * In the case of a setup command (a string), save the computation
```
Comment (cdaringe) [inline src/services/classes/computation-registry.js]: how you feel about making setup of type `Computation`?  just like `Computation` may have a `next` attr, which is _itself a computation_, we could dog food on our Computation runner

Comment (swashcap) [inline src/services/classes/computation-registry.js]: How would we implement this? Joi doesn’t instantiate models. Do it in the `DistributedComputation` constructor?

Comment (cdaringe) [inline src/services/classes/computation-registry.js]: we don't have to do it now in this PR.  but i'm thinking that this file itself can import `Computation` and run `Computation.instantiate(model.setup)`

Comment (swashcap) [inline src/services/classes/computation-registry.js]: Sounds like a plan.

**Pull request merged**